src/cli/circom.rs

```diff
@@ -0,0 +1,118 @@
+    bail!("wasm does not support downloading")
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+fn download_circom_binary(path: impl AsRef<Path>) -> Result<Command> {
+    use std::io::Write;
+
+    let url = match std::env::consts::OS {
+        "linux" => "https://github.com/iden3/circom/releases/download/v2.1.6/circom-linux-amd64",
+        "macos" => "https://github.com/iden3/circom/releases/download/v2.1.6/circom-macos-amd64",
+        "windows" => {
+            "https://github.com/iden3/circom/releases/download/v2.1.6/circom-windows-amd64.exe"
+        }
+        _ => {
+            eprintln!("Unsupported OS");
+            std::process::exit(1);
+        }
+    };
+
+    let response = reqwest::blocking::get(url)?.bytes()?;
+    let mut out = fs::File::create(path.as_ref())?;
+    out.write_all(&response)?;
+
+    #[cfg(unix)]
+    fs::set_permissions(path.as_ref(), fs::Permissions::from_mode(0o755))?;
+
+    Ok(Command::new(path.as_ref().as_os_str()))
+}
+
+/// We try to find the circom binary at `<CIRCOM_DIR>/circom`,
+/// where `<CIRCOM_DIR>` can be configured via the config file,
+/// a environment variable, or through a CLI argument, in that order.
+///
+/// We *do not* consider the case where the user already has some
+/// `circom` binary downloaded. The user will have two possibly
+/// conflicting circom binaries floating around. However, things
+/// should be kept separate as Lurk will never touch the user binary
+/// and the user should never manually call the Lurk Circom binary.
+///
+/// Whatever path is chosen, we then test if the `circom` binary
+/// exists. If it does, we return the path. Otherwise we download
+/// the binary to the location and return the path.
+fn get_circom_binary() -> Result<Command> {
+    let circom_path = circom_binary_path();
+
+    let output = Command::new(&circom_path).arg("--version").output();
+
+    let success = match output {
+        Ok(output) => {
+            // TODO: in future add back checksum check?
+            output.status.success()
+                && String::from_utf8_lossy(&output.stdout).contains(CIRCOM_VERSION)
+        }
+        Err(_) => false,
+    };
+
+    if success {
+        Ok(Command::new(circom_path))
+    } else {
+        download_circom_binary(circom_path)
+    }
+}
+
+pub(crate) fn create_circom_gadget(circom_folder: Utf8PathBuf, name: String) -> Result<()> {
+    let circom_gadget = circom_dir().join(&name);
+    let circom_file = circom_folder.join(&name).with_extension("circom");
+
+    println!(
+        "Running circom binary to generate r1cs and witness files to {:?}",
+        circom_gadget
+    );
+    fs::create_dir_all(&circom_gadget)?;
+    let output = get_circom_binary()?
+        .arg(circom_file)
+        .arg("--r1cs")
+        .arg("--wasm")
+        .arg("--output")
+        .arg(&circom_gadget)
+        .arg("--prime")
+        .arg("vesta")
+        .output()
+        .expect("circom failed");
```

Comment:
Are we sure the `vesta` prime is the one we should use? what about bn256 support?
In practice, the CLI has an idea of what field and proving system we’re being called with, and should pass that to you here.

Nit: You can put arg in the same line for concision `.args(&["--r1cs", "--wasm", "--output", "--prime", "vesta"])`